File: blog-sever/server-java/src/main/java/com/site/blog/model/dto/Result.java
package com.site.blog.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;


@Data
@AllArgsConstructor

public class Result<T> implements Serializable {
    @Serial
    private static final long serialVersionUID = 1L;
    private Integer resultCode;
    private String message;
    private T data;
    private Boolean success;
    private long timestamp;

    public Result() {
    }

    public Result(int resultCode, String message) {
        this.resultCode = resultCode;
        this.message = message;
    }


}

File: myblog-kt/src/main/kotlin/com/site/blog/model/dto/Result.kt
package com.site.blog.model.dto

import java.io.Serial
import java.io.Serializable

data class Result<T>  (
    var resultCode: Int? = null,
    var message: String? = null,
    var data: T? = null,

    var success: Boolean? = null,
    var timestamp: Long = 0



)